File: Python/programmers/etc/level2/테이블_해시_함수.py
def solution(data, col, row_begin, row_end):
    answer = 0
    data.sort(key=lambda t: (t[col-1], -t[0]))
    for i in range(row_begin-1, row_end):
        tmp = 0
        for j in range(len(data[0])):
            tmp += data[i][j]%(i+1)
        answer ^= tmp
    return answer


data = [[2,2,6],[1,5,10],[4,2,9],[3,8,3]]
col, row_begin, row_end = 2, 2, 3
print(solution(data, col, row_begin, row_end)) # 4

File: Java/programmers/etc/level2/테이블해시함수.java
package CodeTest.Java.programmers.etc.level2;

import java.util.*;

public class 테이블해시함수 {
    public int solution(int[][] data, int col, int row_begin, int row_end) {
        int answer = 0;
        Arrays.sort(data, (o1,o2) -> o1[col-1]!=o2[col-1] ? o1[col-1]-o2[col-1] : o2[0]-o1[0]);
        
        for (int i=row_begin-1; i<row_end; i++) {
            int tmp = 0;
            for (int j=0; j<data[0].length; j++) {
                tmp += data[i][j]%(i+1);
            }
            answer ^= tmp;
        }
        
        return answer;
    }
}
